tests/integration/targets/ec2_launch_template/playbooks/roles/ec2_launch_template/tasks/instance-metadata.yml

```diff
@@ -0,0 +1,55 @@
+- block:
+  - name: metadata_options
+    ec2_launch_template:
+      name: "{{ resource_prefix }}-test-metadata"
+      metadata_options:
+        http_put_response_hop_limit: 1
+        http_tokens: required
+      state: present
+    register: metadata_options_launch_template
+    ignore_errors: true
+  - assert:
+      that:
+        - metadata_options_launch_template is not failed
```

Comment:
```suggestion
```

Let a failure trigger the always block and fail, there's no need to explicitly ignore_errors and then assert not failed.  The always block will trigger the cleanup with a failure at any point in these tests